models/running_output.go

```diff
@@ -207,8 +207,22 @@ func (r *RunningOutput) Close() {
 }
 
 // AddMetric adds a metric to the output.
-// Takes ownership of metric
+// The given metric will be copied if the output selects the metric.
 func (r *RunningOutput) AddMetric(metric telegraf.Metric) {
+	ok, err := r.Config.Filter.Select(metric)
+	if err != nil {
+		r.log.Errorf("filtering failed: %v", err)
+	} else if !ok {
+		r.MetricsFiltered.Incr(1)
```

Comment:
Looks great! Just one minor nitpick: is there any reason this isn't calling `r.metricFiltered(metric)` like the similar functions?

Reply:
Yes, that function also calls Drop on the metric, which we should not do if we haven't copied it yet, since we haven't taken ownership of it until then.